src/components/tabs/js/tabsController.js

```diff
@@ -635,17 +635,36 @@ function MdTabsController ($scope, $element, $window, $mdConstant, $mdTabInkRipp
         newHeight     = contentHeight + tabsHeight,
         currentHeight = $element.prop('clientHeight');
     if (currentHeight === newHeight) return;
+
+    // Lock during animation so the user can't change tabs
     locked = true;
-    $animate
-        .animate(
-        $element,
-        { height: currentHeight + 'px' },
-        { height: newHeight + 'px' }
-    )
-        .then(function () {
-                $element.css('height', '');
-                locked = false;
-              });
+
+    var oldCssHeight = { height: currentHeight + 'px'},
+        newCssHeight = { height: newHeight + 'px' };
+
+    // Set the height to the current, specific pixel height to fix a bug on iOS where the height
+    // first animates to 0, then back to the proper height causing a visual glitch
+    $element.css(oldCssHeight);
+
+    // Animate the height from the old to the new
+    $animate.animate($element, oldCssHeight, newCssHeight).then(function () {
```

Comment:
It was using `$animate` so I didn't want to alter the existing code too much, but I'll give this a try and run all of the tests.